private/model/api/endpoint_arn.go

```diff
@@ -23,13 +23,15 @@ const endpointARNShapeTmplDef = `
 
 		// updateArnableField updates the value of the input field that 
 		// takes an ARN as an input. This method is useful to backfill 
-		// the parsed resource name from ARN into the input member. 
-		func (s *{{ $.ShapeName }}) updateArnableField(v string) error {
+		// the parsed resource name from ARN into the input member.
+		// It returns a pointer to a modified copy of input and an error.
+		// Note that original input is not modified. 
+		func (s {{ $.ShapeName }}) updateArnableField(v string) (interface{}, error) {
 			if s.{{ $name }} == nil {
-				return fmt.Errorf("member {{ $name }} is nil")
+				return nil, fmt.Errorf("member {{ $name }} is nil")
 			}
 			s.{{ $name }} = aws.String(v)
-			return nil
+			return &s, nil 
```

Comment:
Should this create a copy of `s` before assigning the value, then return the copy?

Reply:
Missed this was changed to value instead of pointer receiver.